internal/auth/auth.go

```diff
@@ -0,0 +1,62 @@
+package auth
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"os"
+
+	"github.com/flightctl/flightctl/internal/auth/authn"
+	"github.com/flightctl/flightctl/internal/auth/authz"
+	"github.com/flightctl/flightctl/internal/config"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// DisableAuthEnvKey is the environment variable key used to disable auth when developing.
+	DisableAuthEnvKey = "FLIGHTCTL_DISABLE_AUTH"
+)
+
+type AuthNMiddleware interface {
+	AuthHandler(next http.Handler) http.Handler
+}
+
+type AuthZMiddleware interface {
+	CheckPermission(ctx context.Context, resource string, op string) (bool, error)
+}
+
+var auth AuthZMiddleware
+
+func GetAuth() AuthZMiddleware {
+	return auth
+}
+
+func CreateAuthMiddleware(cfg *config.Config, log logrus.FieldLogger) (AuthNMiddleware, error) {
+	value, exists := os.LookupEnv(DisableAuthEnvKey)
+	if exists && value != "" {
+		log.Println("Auth disabled")
```

Comment:
Should this be a more prominent warning like https://github.com/flightctl/flightctl/blob/main/internal/agent/config.go#L119?

Reply:
yes, that would be better. It is now a warning.